packages/foam-core/src/markdown-provider.ts

```diff
@@ -143,45 +157,78 @@ export function createMarkdownReferences(
     return [];
   }
 
-  return graph
-    .getForwardLinks(noteId)
-    .map(link => {
-      let target = graph.getNote(link.to);
-      // if we don't find the target by ID we search the graph by slug
-      if (!target) {
-        const candidates = graph.getNotes({ slug: link.link.slug });
-        if (candidates.length > 1) {
+  // if note doesn't exist, we can't find its links
+  const note = graph.getNote(noteId);
+  if (!note) {
+    return [];
+  }
+
+  const forwardLinks = graph.getForwardLinks(noteId);
+
+  // Try to generate a definition for each [[link]] in the note.
+  //
+  // A note may have multiple [[link]] expressions to the same target
+  // note with different aliases.s
+  //
+  // - note.links contains all the [[link]] expressions
+  // - forwardLinks contains all the edges (links) between documents
+  return (
+    note.links
```

Comment:
shall we make these unique by `literalContent`? We should probably add a test just in case

e.g.
```
that was [[a-great-day]] and anytime I have [[a-great-day]] things are good.
```

(if this turns out to be a bug I guess it was already present in the existing code)

Reply:
Yes, we should add it here, good catch! Right now the unique check is done on the consumer end after stringifying them, but it makes sense to move over here. Will do!